lib/compilers/csharp.js

```diff
@@ -0,0 +1,155 @@
+
+        const projectFilePath = path.join(programDir, 'Program.csproj');
+        const coreClrJitPath = path.join(clrBuildDir, 'libclrjit.so');
+        const crossgenPath = path.join(clrBuildDir, 'crossgen');
+
+        const programPublishPath = path.join(
+            programDir,
+            'bin',
+            this.buildConfig,
+            this.targetFramework,
+            this.rID,
+            'publish',
+        );
+
+        const programDllPath = path.join(programPublishPath, 'Program.dll');
+
+        const projectFileContent =
+        `<Project Sdk="Microsoft.NET.Sdk">
+            <PropertyGroup>
+         <TargetFramework>${this.targetFramework}</TargetFramework>
+         <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+         </PropertyGroup>
+         </Project>
+        `;
+
+        execOptions.customCwd = programDir;
+        await fs.writeFile(projectFilePath, projectFileContent);
+        options = this.compilerOptions;
+
+        const compilerResult = await super.runCompiler(
+            compiler,
+            options,
+            inputFileName,
+            execOptions,
+        );
+
+        if (compilerResult.code !== 0)
+            return compilerResult;
+
+        const crossgenResult = await this.runCrossgen(
+            execOptions,
+            crossgenPath,
+            coreClrJitPath,
+            programPublishPath,
+            programDllPath,
+            this.getOutputFilename(programDir, ''),
+        );
+
+        if (crossgenResult.code !== 0)
+            return crossgenResult;
+
+        return compilerResult;
+    }
+
+    optionsForFilter() {
+        return this.compilerOptions;
+    }
+
+    getOutputFilename(dirPath) {
+        return path.join(dirPath, `output.s`);
+    }
+
+    cleanAsm(asm) {
+        var cleanedAsm = '';
+        var asmLines = asm.split('\n');
+
+        for (const line of asmLines) {
+            if (line.startsWith('; Assembly listing for method')) {
+                // ; Assembly listing for method ConsoleApplication.Program:Main(System.String[])
+                //                               ^ This character is the 31st character in this string.
+                // `substring` removes the first 30 characters from it and uses the rest as a label.
+                cleanedAsm = cleanedAsm.concat(line.substring(30) + ':\n');
+                continue;
+            }
+
+            if (line.startsWith('Microsoft (R)') ||
+                line.startsWith('Copyright (c)') ||
+                line.startsWith('Native image'))
+            {
+                continue;
+            }
+
+            // Removes the raw opcodes from crossgen's output.
+            cleanedAsm = cleanedAsm.concat(line.replace(/ +[\dA-Z]+ +[^\dA-Za-z]/,' ') + '\n');
+        }
+
+        return cleanedAsm;
+    }
+
+    async runCrossgen(execOptions, crossgenPath, jitPath, publishPath, dllPath, outputPath) {
+        execOptions.env['COMPlus_NgenDisasm'] = '*';
```

Comment:
If `crossgen2` is used, need to change this to `--codegenopt NgenDisasm=* --codegenopt NgenDiffableDasm=1`